app/src/main/java/com/pramod/dailyword/framework/ui/worddetails/word_history/SpannableParser.kt

```
@@ -0,0 +1,118 @@
+ * - `[u]text[/u]`: Underlines text.
+ * - `[s]text[/s]`: Strikes through text.
+ * - `[size=20]text[/size]`: Sets font size in sp.
+ * - `[color=#FFFFFF,#000000]text[/color]`: Sets text color. Accepts a comma-separated pair for light and dark modes.
+ * - `[link url="https://example.com"]text[/link]`: Creates a clickable link.
+ *
+ * @param linkColor The color to be applied to the text within `[link]` tags.
+ * @param isSystemInDarkTheme Determines which color to pick from the `[color]` tag's parameter list.
+ * @param onLinkClick A callback invoked when a clickable link is tapped, providing the URL string.
+ * @return An [AnnotatedString] with the styles and annotations applied.
+ */
+suspend fun String.parseToCustomAnnotatedString(
```

detekt (1.23.8)

[warning] The function parseToCustomAnnotatedString appears to be too complex based on Cognitive Complexity (complexity: 41). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)